src/compiler/parser.ts

```diff
@@ -10517,9 +10517,10 @@ export function getDeclarationFileExtension(fileName: string): string | undefine
         return standardExtension;
     }
     if (fileExtensionIs(fileName, Extension.Ts)) {
-        const index = getBaseFileName(fileName).lastIndexOf(".d.");
+        const baseName = getBaseFileName(fileName);
+        const index = baseName.lastIndexOf(".d.");
         if (index >= 0) {
-            return fileName.substring(index);
+            return baseName.substring(index);

```

Comment:
This was extremely bad; I guess we were only really using this to determine whether something _was_ a declaration file name, not to actually use the extension.